### PR TITLE
BUG: xticks unnecessarily rotated

### LINKED ISSUE #29460: xticks unnecessarily rotated
#### Code Sample
```python
import pandas as pd
import matplotlib.pyplot as plt
from io import StringIO

csv_string = """x,y1,y2
0, 1, 1
1, 1, 2
2, 1, 3
3, 4, 2
4, 2, 4
5, 3, 3
"""

data = pd.read_csv(StringIO(csv_string))
dataframe = pd.DataFrame(data)

dataframe.plot(x="x",
               y=["y1", "y2"],
               kind="line",
               subplots=True,
               sharex=True,
               marker=",",
               linewidth=0.5,
               title="Title"
               )

plt.show()
```
#### Problem description
The xticks are unnecessarily rotated:
grafik
Also the title overlaps the subplot, but I will create a new issue for that.

#### Expected Output
The xticks shouldn't be rotated.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.2.final.0
python-bits      : 32
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.3
numpy            : 1.17.3
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.6.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
import sys; print('Python %s on %s' % (sys.version, sys.platform))
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 22:20:52) [MSC v.1916 32 bit (Intel)] on win32
</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] closes #29460
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

On `master`, xticklabels are rotated if
```python
            not self._use_dynamic_x()
            and data.index.is_all_dates
            and not self.subplots
            or (self.subplots and self.sharex)
```

There's a comment saying
```python
        if condition:
            # irregular TS rotated 30 deg. by default
            # probably a better place to check / set this.
            if not self._rot_set:
                self.rot = 30
```
, and it seems that the _intention_ was to rotate ticklabels if there's an irregular timeseries.

With the condition as it is written, they will also be rotated if the following is true:
```python
(self.subplots and self.sharex)
```

I don't see why that should warrant rotating the ticklabels, so I'd like to suggest that there was a bracketing error.

Also, the condition `data.index.is_all_dates` is (as far as I can tell) only relevant if `use_index=True`, so I've changed that.

Here's what the OP's plot looks like with these changes:

image

Here's what it looks like on master:

image

Irregular time series still have their ticklabels rotated:
image

Regular time series' ticklabels don't rotate:
image

(unless you pass `xcompat=True`, but that's unaffected by this change)

### CODE REVIEW

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -1204,9 +1204,8 @@ def get_label(i):
 
         condition = (
             not self._use_dynamic_x()
-            and data.index.is_all_dates
-            and not self.subplots
-            or (self.subplots and self.sharex)
+            and (data.index.is_all_dates and self.use_index)
+            and (not self.subplots or (self.subplots and self.sharex))
```
Comment (charlesdong1991) [inline pandas/plotting/_matplotlib/core.py:1208]: could you pls add descriptive comment for `condition`? seems quite complex now

Comment (jreback) [reply pandas/plotting/_matplotlib/core.py:1208]: yeah this is getting very complex, pls comment , or use a helper variable to make this more clear / readable.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -1202,11 +1202,15 @@ def get_label(i):
...
+        # subplots which don't share their x-axes, in which we case
+        # we don't rotate the ticklabels as by default the subplots
+        # would be too close together.
         condition = (
             not self._use_dynamic_x()
-            and data.index.is_all_dates
-            and not self.subplots
-            or (self.subplots and self.sharex)
+            and (data.index.is_all_dates and self.use_index)
+            and (not self.subplots or (self.subplots and self.sharex))
```
Comment (MarcoGorelli) [inline pandas/plotting/_matplotlib/core.py:1213]: note to self: add a test which hits
```python
and (not self.subplots or (self.subplots and self.sharex))
```

**Review by charlesdong1991 — COMMENTED** [review]
very nice! 

only one nitpick, but could also choose to ignore since it doesn't really matter

File: pandas/io/formats/info.py [hunk 3]
```diff
@@ -39,6 +39,37 @@ def _put_str(s: Union[str, Dtype], space: int) -> str:
     return str(s)[:space].ljust(space)
 
 
+def _sizeof_fmt(num: Union[int, float], size_qualifier: str) -> str:
```
Comment (charlesdong1991) [inline pandas/io/formats/info.py:42]: seems the change in this file is by accident added ^^

File: pandas/tests/plotting/test_datetimelike.py [hunk 4]
```diff
@@ -1491,6 +1491,34 @@ def test_matplotlib_scatter_datetime64(self):
...
+
+        # index of dates but don't use index
+        axes = df.set_index("x").plot(y="y", use_index=False)
+        self._check_ticks_props(axes, xrot=0)
+
+        # regular time series
+        x = to_datetime(["2020-05-01", "2020-05-02", "2020-05-03"])
+        df = DataFrame({"x": x, "y": [1, 2, 3]})
+        axes = df.plot(x="x", y="y")
+        self._check_ticks_props(axes, xrot=0)
```
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply pandas/tests/plotting/test_datetimelike.py]: That's a good suggestion, thanks!

File: pandas/plotting/_matplotlib/core.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/plotting/_matplotlib/core.py:1217]: Sure, done

**Review by jreback — CHANGES_REQUESTED** [review]
can u merge master and ping on green

**Comment by MarcoGorelli** [comment]
@jreback ping

File: doc/source/whatsnew/v1.1.0.rst [hunk 5]
```diff
@@ -1099,6 +1099,7 @@ Plotting
 - Bug in :meth:`DataFrame.boxplot` and :meth:`DataFrame.plot.boxplot` lost color attributes of ``medianprops``, ``whiskerprops``, ``capprops`` and ``medianprops`` (:issue:`30346`)
 - Bug in :meth:`DataFrame.hist` where the order of ``column`` argument was ignored (:issue:`29235`)
 - Bug in :meth:`DataFrame.plot.scatter` that when adding multiple plots with different ``cmap``, colorbars alway use the first ``cmap`` (:issue:`33389`)
+- Bug in :meth:`Dataframe.plot` was rotating xticklabels when subplots was equal to True, even if the x-axis wasn't an irregular time series (:issue:`29460`)
```
Comment (charlesdong1991) [inline doc/source/whatsnew/v1.1.0.rst:1102]: i am very sorry to be a badass to block your PR, can you change `Dataframe` to `DataFrame`? 😅

Comment (charlesdong1991) [reply doc/source/whatsnew/v1.1.0.rst:1102]: and also double quote `subplots`
```suggestion
- Bug in :meth:`DataFrame.plot` was rotating xticklabels when ``subplots`` was equal to True, even if the x-axis wasn't an irregular time series (:issue:`29460`)
```

Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.1.0.rst:1102]: :laughing: good catch, thanks!

**Comment by simonjayhawkins** [comment]
@MarcoGorelli can you move release note to 1.2

File: doc/source/whatsnew/v1.2.0.rst [hunk 6]
```diff
@@ -239,7 +239,7 @@ I/O
 Plotting
 ^^^^^^^^
 
--
+- Bug in :meth:`DataFrame.plot` was rotating xticklabels when subplots was equal to True, even if the x-axis wasn't an irregular time series (:issue:`29460`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:242]: use ```subplots=True``` (in double-backticks) so its easily readable

**Review by jreback — CHANGES_REQUESTED** [review]
tiny comment, otherwsie lgtm

**Comment by jreback** [comment]
@charlesdong1991 @jorisvandenbossche @TomAugspurger if comments.

**Comment by TomAugspurger** [comment]
Nothing here.

**Review by charlesdong1991 — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
@MarcoGorelli can you rebase.  looks like this was about ready

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks for this @MarcoGorelli nice! (and good to add the testing like you did).

**Pull request merged**